### PR TITLE
Add trigger manager and cron trigger

### PR DESCRIPTION
**Pull request by AbdelrahmanElawady**
## Description 

Add trigger manager and some methods to interact with triggers. Of course it needs the scheduler as part of its operation but this is just a basic implementation.

## Issues

closes: #24

### CODE REVIEW

File: triggers/cron_test.go
Comment (KhaledHegazy222) [inline triggers/cron_test.go:1]: why did you only add tests for validating cron jobs but there is no tests for registering and canceling jobs ?

**Review by KhaledHegazy222 — COMMENTED** [review]
thank you for you effort but I just have some comments about the cron jobs trigger implementation and tests

File: triggers/cron.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	cronScheduler *gocron.Scheduler
+	jobs          map[int]*gocron.Job
+	// workflowScheduler *scheduler.Scheduler
+}
+
+const cronTrigger = "cron"
+
+// const cronMetadata = ""
+
+var _ handler = (*cronHandler)(nil)
```
[the remark this answers is not in the document]
Comment (AbdelrahmanElawady) [reply triggers/cron.go:23]: This is a way to ensure a struct implement an interface at compile time without memory allocation. [reference](https://stackoverflow.com/questions/10498547/ensure-a-type-implements-an-interface-at-compile-time-in-go)

File: triggers/cron.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	c.jobs[id] = job
+	return nil
+}
+
+func (c *cronHandler) cancel(id int) error {
+	job, ok := c.jobs[id]
+	if !ok {
+		return ErrWorkflowNotFound
+	}
+	c.cronScheduler.Remove(job)
```
[the remark this answers is not in the document]
Comment (AbdelrahmanElawady) [reply triggers/cron.go:65]: I forgot that will add it

File: triggers/cron_test.go
Comment (AbdelrahmanElawady) [reply triggers/cron_test.go:1]: registering is still waiting on scheduler but sure will add for canceling

File: triggers/cron.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (KhaledHegazy222) [reply triggers/cron.go:23]: got it, nice

**Review by KhaledHegazy222 — DISMISSED** [review]

**Review by Omar622 — APPROVED** [review]
nice work

**Review by KhaledHegazy222 — APPROVED** [review]

**Pull request merged**